code/modules/mob/living/simple_animal/hostile/ooze.dm

```diff
@@ -0,0 +1,490 @@
+
+///Remove the speed modifier and delete the timer for heating up
+/datum/action/cooldown/metabolicboost/proc/FinishSpeedup(timerid)
+	var/mob/living/simple_animal/hostile/ooze/ooze = owner
+	ooze.remove_movespeed_modifier(/datum/movespeed_modifier/metabolicboost)
+	to_chat(ooze, "<span class='notice'>You start slowing down again.</span>")
+	deltimer(timerid)
+	active = FALSE
+	StartCooldown()
+
+
+///This action lets you consume the mob you're currently pulling. I'M GONNA CONSUUUUUME (this is considered one of the funny memes in the 2019-2020 era)
+/datum/action/consume
+	name = "Consume"
+	desc = "Consume a mob that you are dragging to gain nutrition from them"
+	background_icon_state = "bg_hive"
+	icon_icon = 'icons/mob/actions/actions_slime.dmi'
+	button_icon_state = "consume"
+	check_flags = AB_CHECK_CONSCIOUS|AB_CHECK_STUN
+	///The mob thats being consumed by this creature
+	var/mob/living/vored_mob
+
+///Register for owner death
+/datum/action/consume/New(Target)
+	. = ..()
+	RegisterSignal(owner, COMSIG_MOB_DEATH, .proc/on_owner_death)
+
+///Try to consume the pulled mob
+/datum/action/consume/Trigger()
+	. = ..()
+	if(!.)
+		return
+	var/mob/living/simple_animal/hostile/ooze/gelatinous/ooze = owner
+	if(!ooze.pulling)
+		to_chat(src, "<span class='warning'>You need to be pulling a creature for this to work!</span>")
+		return FALSE
+	if(!isliving(ooze.pulling))
+		to_chat(src, "<span class='warning'>You need to be pulling a creature for this to work!</span>")
+		return FALSE
+	if(vored_mob)
+		to_chat(src, "<span class='warning'>You are already consuming another creature!</span>")
+		return FALSE
+	owner.visible_message("<span class='warning>[ooze] starts attempting to devour [target]!</span>", "<span class='notice'>You start attempting to devour [target].</span>")
+	if(!do_after(ooze, 15, target = ooze.pulling))
+		return FALSE
+	var/mob/living/eat_target = ooze.pulling
+
+	if(!(eat_target.mob_biotypes & MOB_ORGANIC) || eat_target.stat == DEAD)
+		to_chat(src, "<span class='warning'>This creature isn't to my tastes!</span>")
+		return FALSE
+	start_consuming(eat_target)
+
+///Start allowing this datum to process to handle the damage done to  this mob.
+/datum/action/consume/proc/start_consuming(mob/living/target)
+	vored_mob = target
+	vored_mob.forceMove(owner) ///AAAAAAAAAAAAAAAAAAAAAAHHH!!!
+	playsound(owner,'sound/items/eatfood.ogg', rand(30,50), TRUE)
+	owner.visible_message("<span class='warning>[src] devours [target]!</span>", "<span class='notice'>You devour [target].</span>")
+	START_PROCESSING(SSprocessing, src)
+
+///Stop consuming the mob; dump them on the floor
+/datum/action/consume/proc/stop_consuming()
+	STOP_PROCESSING(SSprocessing, src)
+	vored_mob.forceMove(get_turf(owner))
+	playsound(get_turf(owner), 'sound/effects/splat.ogg', 50, TRUE)
+	owner.visible_message("<span class='warning>[owner] pukes out [vored_mob]!</span>", "<span class='notice'>You puke out [vored_mob].</span>")
+	vored_mob = null
```

Comment:
What if the mob gets qdeleted in between?

Reply:
will add sanity check

Reply:
More than a sanity check, register for its deletion and handle it appropriately.